src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java

```diff
@@ -25,28 +45,107 @@
+                                      ApplicationProperties applicationProperties) {
         this.oneTimePasswordRepository = oneTimePasswordRepository;
         this.oneTimePasswordMapper = oneTimePasswordMapper;
+        this.otpSpecService = otpSpecService;
+        this.loadBalancedRestTemplate = template;
+        this.communicationService = communicationService;
+        this.applicationProperties = applicationProperties;
     }
 
     /**
      * Save a oneTimePassword.
      *
-     * @param oneTimePasswordDTO the entity to save
+     * @param oneTimePasswordDTO the entity to generate
      * @return the persisted entity
      */
     @Override
-    public OneTimePasswordDTO save(OneTimePasswordDTO oneTimePasswordDTO) {
-        log.debug("Request to save OneTimePassword : {}", oneTimePasswordDTO);
-        OneTimePassword oneTimePassword = oneTimePasswordMapper.toEntity(oneTimePasswordDTO);
-        oneTimePassword = oneTimePasswordRepository.save(oneTimePassword);
-        return oneTimePasswordMapper.toDto(oneTimePassword);
+    @Transactional
+    public OneTimePasswordDTO generate(OneTimePasswordDTO oneTimePasswordDTO) {
+        log.debug("Request to generate OneTimePassword : {}", oneTimePasswordDTO);
+        OtpSpec.OtpTypeSpec oneType = getOneTypeSpec(oneTimePasswordDTO.getTypeKey());
+
+        //generate otp
+        Generex generex = new Generex(oneType.getPattern());
+        String randomPasswrd = generex.random();
+
+        //build domain
+        OneTimePassword otp = getOneTimePassword(oneTimePasswordDTO, oneType, randomPasswrd);
+
+        String message = oneType.getMessage().getEn().replaceAll("\\$\\{otp}", randomPasswrd);
+        oneTimePasswordRepository.saveAndFlush(otp);
+
+        sendOneTimePassword(message, otp.getReceiver(), oneType.getOtpSenderId());
+        return oneTimePasswordMapper.toDto(otp);
+    }
+
+    private void sendOneTimePassword(String message, String receiver, String senderId) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(HttpHeaders.AUTHORIZATION, communicationService.getSystemToken());
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        String url = applicationProperties.getCommunicationUrl()
+            + "/tmf-api/communicationManagement/v2/communicationMessage/send";
+        CommunicationMessage body = new CommunicationMessage();
+        body.setContent(message);
+        body.setType("SMS");
+        body.setSender(new Sender(senderId));
+        body.setReceiver(new ArrayList<>());
+        body.getReceiver().add(new Receiver(receiver, receiver));
+        RequestEntity<Object> request = new RequestEntity<>(body, headers, POST, URI.create(url));
+        loadBalancedRestTemplate.exchange(request, Object.class);
+    }
+
+    private OneTimePassword getOneTimePassword(
+        OneTimePasswordDTO oneTimePasswordDTO,
+        OtpSpec.OtpTypeSpec oneType,
+        String randomPasswrd
+    ) {
```

Comment:
wrong formating
...
 getOneTimePassword(OneTimePasswordDTO oneTimePasswordDTO,
...
 String randomPasswrd) {

expected, also please check in another methods